polaris-react/src/components/DataTable/DataTable.scss

```diff
@@ -337,3 +329,19 @@
     box-shadow: var(--p-shadow-base);
   }
 }
+
+.FixedFirstColumn {
+  position: absolute;
+
+  background: white;
+  @media #{$p-breakpoints-md-down} {
+    z-index: 1;
+  }
+
+  z-index: 3;
```

Comment:
Sorry, I can't remember why this had to be higher than on mobile. Can we just use `z-index: 1` here, as well? This way the scrollbar remains visible rather than getting cut-off by the fixed column.

https://screenshot.click/24-05-821ic-15bqm.mp4